AppBundles/RFAExportRCEPlugin/RFAExportRCEAutomation.cs

```diff
@@ -0,0 +1,177 @@
+// MERCHANTABILITY OR FITNESS FOR A PARTICULAR USE.  AUTODESK, INC.
+// DOES NOT WARRANT THAT THE OPERATION OF THE PROGRAM WILL BE
+// UNINTERRUPTED OR ERROR FREE.
+/////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+using Inventor;
+using Autodesk.Forge.DesignAutomation.Inventor.Utils;
+
+namespace RFAExportRCEPlugin
+{
+    [ComVisible(true)]
+    public class RFAExportRCEAutomation
+    {
+        private readonly InventorServer inventorApplication;
+
+        public RFAExportRCEAutomation(InventorServer inventorApp)
+        {
+            inventorApplication = inventorApp;
+        }
+
+        public void Run(Document doc)
+        {
+            LogTrace("Run called with {0}", doc.DisplayName);
+            try
+            {
+                using (new HeartBeat())
+                {
+                    ExportRFA(doc);
+                }
+            }
+            catch (Exception e)
+            {
+                LogError("Processing failed. " + e.ToString());
+            }
+        }
+
+        public void RunWithArguments(Document doc, NameValueMap map)
+        {
+
+        }
+
+        BIMComponent getBIMComponent(Document doc)
+        {
+            BIMComponent bimComponent = null;
+            var docType = doc.DocumentType;
+            if (docType == DocumentTypeEnum.kAssemblyDocumentObject)
+            {
+                AssemblyDocument _doc = doc as AssemblyDocument;
+                bimComponent = _doc.ComponentDefinition.BIMComponent;
+            }
+            else if (docType == DocumentTypeEnum.kPartDocumentObject)
+            {
+                PartDocument _doc = doc as PartDocument;
+                bimComponent = _doc.ComponentDefinition.BIMComponent;
+            }
+            else
+            {
+                Trace.TraceInformation("NOT supported document type.");
+            }
+
+            return bimComponent;
+        }
+
+        #region ExportRFA file 
+
+        public void ExportRFA(Document doc) 
+        { 
+            LogTrace("Export RFA file.");
+
+            BIMComponent bimComponent = getBIMComponent(doc);
+            if (bimComponent == null)
+            {
+                return;
+            }
+
+            var startDir = System.IO.Directory.GetCurrentDirectory();
+
+            // output file name
+            var fileName = System.IO.Path.Combine(startDir, "Output.rfa");
+
+            NameValueMap nvm = inventorApplication.TransientObjects.CreateNameValueMap();
+            LogTrace($"Exporting to {fileName}");
+
+            var reportFileName = System.IO.Path.Combine(startDir, "Report.html");
+
+            LogTrace($"Reporting to {reportFileName}");
+            nvm.Add("ReportFileName", reportFileName);
+
+            DateTime t = DateTime.Now;
+            using (new HeartBeat())
+            {
+                try
```

Comment:
So what will happen if it will not be successful? Will the user be notified? It seems like not.

Reply:
I think that there was missing rfa when I have different output than activity needs as output and then it ends with UploadFailed